internal/provider/integration_azure_resource.go

```diff
@@ -0,0 +1,260 @@
+	resp.TypeName = req.ProviderTypeName + "_integration_azure"
+}
+
+func (r *integrationAzureResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"space_id": schema.StringAttribute{
+				MarkdownDescription: "Mondoo Space Identifier.",
+				Required:            true,
+			},
+			"mrn": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Integration identifier",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Name of the integration.",
+				Optional:            true,
+			},
+			"client_id": schema.StringAttribute{
+				MarkdownDescription: "Azure Client ID.",
+				Required:            true,
+			},
+			"tenant_id": schema.StringAttribute{
+				MarkdownDescription: "Azure Tenant ID.",
+				Required:            true,
+			},
+			"scan_vms": schema.BoolAttribute{
+				MarkdownDescription: "Scan VMs.",
+				Optional:            true,
+			},
+			"subscription_whitelist": schema.ListAttribute{
+				MarkdownDescription: "List of Azure subscriptions to scan.",
+				Optional:            true,
+				ElementType:         types.StringType,
+			},
+			"subscription_blacklist": schema.ListAttribute{
+				MarkdownDescription: "List of Azure subscriptions to exclude from scanning.",
+				Optional:            true,
+				ElementType:         types.StringType,
+			},
+			"credentials": schema.SingleNestedAttribute{
+				Required: true,
+				Attributes: map[string]schema.Attribute{
+					"pem_file": schema.StringAttribute{
+						MarkdownDescription: "PEM file for Azure integration.",
+						Required:            true,
+						Sensitive:           true,
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *integrationAzureResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*mondoov1.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = &ExtendedGqlClient{client}
+}
+
+func (r *integrationAzureResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+
+	var data integrationAzureResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Do GraphQL request to API to create the resource.
+	spaceMrn := ""
+	if data.SpaceId.ValueString() != "" {
+		spaceMrn = spacePrefix + data.SpaceId.ValueString()
+	}
+
+	var listWhite []mondoov1.String
```

Comment:
we should avoid black and white as terms for lists, better to stick with allow and deny list.